go/enclave/rollupmanager/rollupmanager.go

```diff
@@ -139,14 +138,16 @@ func (re *rollupManager) CreateRollup() (*core.Rollup, error) {
 	return newRollup, nil
 }
 
-func (re *rollupManager) ProcessL1Block(b *types.Block) ([]*core.Rollup, error) {
-	return re.processRollups(b)
+func (re *rollupManager) ProcessL1Block(br *common.BlockAndReceipts) ([]*core.Rollup, error) {
+	return re.processRollups(br)
 }
 
 // extractRollups - returns a list of the rollups published in this block
-func (re *rollupManager) extractRollups(b *types.Block, blockResolver db.BlockResolver) []*core.Rollup {
+func (re *rollupManager) extractRollups(br *common.BlockAndReceipts, blockResolver db.BlockResolver) []*core.Rollup {
 	rollups := make([]*core.Rollup, 0)
-	for _, tx := range b.Transactions() {
+	b := br.Block
+
+	for _, tx := range *br.SuccessfulTransactions() {
```

Comment:
how many times do we iterate over the successful transactions?
If more than once, it would be worth using a map data structure in `BlockAndReceipts` to back the receicpts
`Map<Contract_Address,[]Receipt>`